src/pkg_manager.ts

```diff
@@ -16,17 +17,44 @@ function modeToFlag(mode: InstallOptions["mode"]): string {
     : "";
 }
 
+function modeToFlagYarn(mode: InstallOptions["mode"]): string {
+  return mode === "dev"
+    ? "--dev"
+    : mode === "optional"
+    ? "--optional"
+    : "";
+}
```

Comment:
yarn berry does not respect the `--save-dev` and `--save-optional` flags unlike npm + pnpm. bun also has the same problem, it only understands the yarn style `--dev` and `--optional` flags but it's arguably worse than yarn as it doesn't throw errors when an unrecognized flag is passed in. additionally, if bun supported the shorthand for `--optional` (`-O`) then we wouldn't need this at all as all the package managers currently understand the shorthand form for installing a dev dependency (`-D`).